src/thorlabs_cube/driver/kcube/ksc.py

```diff
@@ -0,0 +1,277 @@
+
+        # Send the request to get the current trigger I/O configuration
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_KCUBETRIGIOCONFIG,
+            [MGMSG.MOT_GET_KCUBETRIGIOCONFIG],
+            data=payload,
+        )
+
+        trig1_mode, trig1_polarity, trig2_mode, trig2_polarity = st.unpack(
+            "<BBBB", get_msg.data[2:]
+        )
+        return trig1_mode, trig1_polarity, trig2_mode, trig2_polarity
+
+    async def set_kcubepostrig_params(
+        self,
+        start_pos_fwd,
+        interval_fwd,
+        num_pulses_fwd,
+        start_pos_rev,
+        interval_rev,
+        num_pulses_rev,
+        pulse_width,
+        num_cycles,
+    ):
+        """Set the KCube post-trigger parameters.
+
+        :param start_pos_fwd: Stage position to start the forward trigger sequence.
+        :param interval_fwd: Interval in encoder counts/microsteps for forward trigger pulses.
+        :param num_pulses_fwd: Number of output pulses during forward move.
+        :param start_pos_rev: Stage position to start the reverse trigger sequence.
+        :param interval_rev: Interval in encoder counts/microsteps for reverse trigger pulses.
+        :param num_pulses_rev: Number of output pulses during reverse move.
+        :param pulse_width: Trigger output pulse width (from 1 μs to 1,000,000 μs).
+        :param num_cycles: Number of forward/reverse cycles.
+        """
+        # Pack the Channel ID and trigger parameters
+        payload = st.pack(
+            "<HLLLLLLLL",
+            _CHANNEL,
+            start_pos_fwd,
+            interval_fwd,
+            num_pulses_fwd,
+            start_pos_rev,
+            interval_rev,
+            num_pulses_rev,
+            pulse_width,
+            num_cycles,
+        )
+
+        await self.send(Message(MGMSG.MOT_SET_KCUBEPOSTRIGPARAMS, data=payload))
+
+    async def get_kcubepostrig_params(self):
+        """Get the KCube post-trigger parameters.
+
+        :return: A tuple containing (start_pos_fwd, interval_fwd,
+        num_pulses_fwd, start_pos_rev, interval_rev, num_pulses_rev,
+        pulse_width, num_cycles).
+        """
+
+        payload = st.pack("<H", _CHANNEL)
+
+        # Send the request to get the current post-trigger parameters
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_KCUBEPOSTRIGPARAMS,
+            [MGMSG.MOT_GET_KCUBEPOSTRIGPARAMS],
+            data=payload,
+        )
+
+        return st.unpack("<LLLLLLLL", get_msg.data[2:])
+
+    class KscSim(TscSim):
+        async def set_kcubemmi_params(
+            self,
+            js_mode: int,
+            js_max_vel: int,
+            js_accn: int,
+            dir_sense: int,
+            preset_pos1: int,
+            preset_pos2: int,
+            preset_pos3: int,
+            disp_brightness: int,
+            disp_timeout: int,
+            disp_dim_level: int,
+            js_sensitivity: int,
+        ):
```

Comment:
Simulator class methods should not be asynchronous. But the signatures should have type hints!